desktop/autoUpdateListeners.js

```diff
@@ -6,14 +6,15 @@ import { ipcConsts } from '../app/vars';
 const subscribeToAutoUpdateListeners = ({ mainWindow }) => {
   autoUpdater.on('error', (error) => mainWindow.webContents.send(ipcConsts.WALLET_UPDATE_ERROR, { error }));
 
-  autoUpdater.once('update-downloaded', () => mainWindow.webContents.send(ipcConsts.DOWNLOAD_UPDATE_COMPLETED));
+  autoUpdater.on('update-downloaded', () => mainWindow.webContents.send(ipcConsts.DOWNLOAD_UPDATE_COMPLETED));
 
   ipcMain.on(ipcConsts.QUIT_AND_UPDATE, () => autoUpdater.quitAndInstall());
 
   ipcMain.on(ipcConsts.CHECK_WALLET_UPDATE, async () => {
     const updateCheckResult: UpdateCheckResult = await autoUpdater.checkForUpdates();
     const isUpdateAvailable = !!updateCheckResult.downloadPromise;
     mainWindow.webContents.send(ipcConsts.CHECK_WALLET_UPDATE_SUCCESS, { isUpdateAvailable });
```

Comment:
response isn't used, remove it and associated code

Reply:
Response for 'update-downloaded'? used in App.js in listenToDownloadUpdates. Or are you referring to something else?

Reply:
CHECK_WALLET_UPDATE_SUCCESS

Reply:
the result is used in Settings page when checking for update.

Reply:
copying store value to state is wrong, see what you've done at App.js, this is the right way. And "await" is unnecessary

Reply:
OK, but in App.js I have a listener that changes the state on complete and in progress sets the store value for download in progress. I need the state to be changed based on whether an update is available so I do need the return value, or check value from store directly. In both cases I'd need the CHECK_WALLET_UPDATE_SUCCESS to return a value stating if an update is available.

Reply:
DOWNLOAD_UPDATE_COMPLETED states that there is an update ready and DOWNLOAD_UPDATE_PROGRESS states that update is downloading. 'download-progress' event will be triggered immediately cause the flag is set to download update whenever there is one. All this makes CHECK_WALLET_UPDATE_SUCCESS redundant